### DISCUSSION TITLE
pnpm fetching nodejs

### DISCUSSION THREAD

**Discussion by starbirdtech383** [pnpm/pnpm#7140] [General]
in our project github runner, i am seeing that pnpm command is doing fetch nodejs inspite of nodejs is already installed 
<img width="669" alt="image" src="https://github.com/pnpm/pnpm/assets/113826417/55391f18-c5e9-476e-8469-f4a5709af5ba">

and this happens for every build and sometimes the download takes longer resulting in timeouts or build fail. 

https://github.com/pnpm/pnpm/blob/main/env/plugin-commands-env/src/node.ts#L67
while stubbling up on this, i came across above line of code and trying to understand whats happening here and have few questions below

Q1. is this the line which is trying to fetch during any pnpm command? 
Q2. why is nodejs directory being checked inside pnpm_home directory and is this the bug or some kind of pre-requisite linking expected to be there and somehow in my setup it isnt? 
Q3. any suggestions on how to fix?

Appreciate any inputs here.

**Comment by zkochan** [comment]
It is happening because you have [use-node-version](https://pnpm.io/npmrc#use-node-version) in your `.npmrc`.

Maybe we could check the already available Node.js version and skip installing it no matter it was installed not by pnpm.

**Reply by starbirdtech383** [reply]
thanks a lot. That worked for us! however having it in there is useful for local development purposes so we ended up stripping off conditionally in the job. 

additional info: 
root cause is also that pnpm/action-setup@v2 action is removing it and then forcing it to install again 

      - name: Setup pnpm
        uses: pnpm/action-setup@v2
        with:
          version: 8.8
